main.py

```diff
@@ -195,56 +329,22 @@ def main(cl_arguments):
     start_time = time.time()
     model = build_model(args, vocab, word_embs, tasks)
     log.info('\tFinished building model in %.3fs', time.time() - start_time)
+    return tasks, pretrain_tasks, target_tasks, vocab, word_embs, model
 
-    # Check that necessary parameters are set for each step. Exit with error if not.
-    steps_log = []
-
-    if not args.load_eval_checkpoint == 'none':
-        assert_for_log(os.path.exists(args.load_eval_checkpoint),
-                       "Error: Attempting to load model from non-existent path: [%s]" %
-                       args.load_eval_checkpoint)
-        assert_for_log(
-            not args.do_pretrain,
-            "Error: Attempting to train a model and then replace that model with one from a checkpoint.")
-        steps_log.append("Loading model from path: %s" % args.load_eval_checkpoint)
-
-    assert_for_log(args.transfer_paradigm in ["finetune", "frozen"],
-                   "Transfer paradigm %s not supported!" % args.transfer_paradigm)
-
-    if args.do_pretrain:
-        assert_for_log(args.pretrain_tasks != "none",
-                       "Error: Must specify at least on training task: [%s]" % args.pretrain_tasks)
-        assert_for_log(
-            args.val_interval %
-            args.bpp_base == 0, "Error: val_interval [%d] must be divisible by bpp_base [%d]" %
-            (args.val_interval, args.bpp_base))
-        steps_log.append("Training model on tasks: %s" % args.pretrain_tasks)
-
-    if args.do_target_task_training:
-        steps_log.append("Re-training model for individual eval tasks")
-        assert_for_log(
-            args.eval_val_interval %
-            args.bpp_base == 0, "Error: eval_val_interval [%d] must be divisible by bpp_base [%d]" %
-            (args.eval_val_interval, args.bpp_base))
-        assert_for_log(len(set(pretrain_tasks).intersection(target_tasks)) == 0
-                       or args.allow_reuse_of_pretraining_parameters
-                       or args.do_pretrain == 0,
-                       "If you're pretraining on a task you plan to reuse as a target task, set\n"
-                       "allow_reuse_of_pretraining_parameters = 1(risky), or train in two steps:\n"
-                       "  train with do_pretrain = 1, do_target_task_training = 0, stop, and restart with\n"
-                       "  do_pretrain = 0 and do_target_task_training = 1.")
-
-    if args.do_full_eval:
-        assert_for_log(args.target_tasks != "none",
-                       "Error: Must specify at least one eval task: [%s]" % args.target_tasks)
-        steps_log.append("Evaluating model on tasks: %s" % args.target_tasks)
-
+def main(cl_arguments):
```

Comment:
Since this is the piece of code that we expect people to edit most often, let's comment it _to death_. Explain every block of code, at least one comment for every two or three lines, unless it's _blindingly_ obvious what's going on.